doc/modules/linear_model.rst

```diff
@@ -897,6 +897,169 @@ to warm-starting (see :term:`Glossary <warm_start>`).
     .. [9] `"Performance Evaluation of Lbfgs vs other solvers"
             <http://www.fuzihao.org/blog/2016/01/16/Comparison-of-Gradient-Descent-Stochastic-Gradient-Descent-and-L-BFGS/>`_
 
+.. _Generalized_linear_regression:
+
+Generalized Linear Regression
+=============================
+
+Generalized Linear Models (GLM) extend linear models in two ways
+[10]_. First, the predicted values :math:`\hat{y}` are linked to a linear
+combination of the input variables :math:`X` via an inverse link function
+:math:`h` as
+
+.. math::    \hat{y}(w, x) = h(x^\top w) = h(w_0 + w_1 x_1 + ... + w_p x_p).
+
+Secondly, the squared loss function is replaced by the unit deviance :math:`d`
+of a reproductive exponential dispersion model (EDM) [11]_. The minimization
+problem becomes
+
+.. math::    \min_{w} \frac{1}{2 \sum_i s_i} \sum_i s_i \cdot d(y_i, \hat{y}(w, x_i)) + \frac{\alpha}{2} ||w||_2
```

Comment:
Since we leave out the sample weights in the other formulas on this page, I would do the same here as well. It makes it easier to compare the different formulas.

Reply:
Without it, a user has to look into the source code in order to see what excactly happens when sample weights are specified. I‘d rather add sample weights to ridge in the user guide. But that‘s just my oppinion.

Reply:
I agree the way we SWs is really poorly documented in general :/

I'd be fine with both formula: first without SW, then with SW